src/contracts/middleware/BLSIndexRegistryCoordinator.sol

```diff
@@ -0,0 +1,163 @@
+        IServiceManager _serviceManager,
+        IBLSPubkeyRegistry _blsPubkeyRegistry,
+        IIndexRegistry _indexRegistry
+    ) StakeRegistry(_strategyManager, _serviceManager) {
+        blsPubkeyRegistry = _blsPubkeyRegistry;
+        indexRegistry = _indexRegistry;
+    }
+
+    function initialize(
+        uint96[] memory _minimumStakeForQuorum,
+        StrategyAndWeightingMultiplier[][] memory _quorumStrategiesConsideredAndMultipliers
+    ) external override initializer {
+        // the stake registry is this contract itself
+        registries.push(address(this));
+        registries.push(address(blsPubkeyRegistry));
+        registries.push(address(indexRegistry));
+
+        StakeRegistry._initialize(_minimumStakeForQuorum, _quorumStrategiesConsideredAndMultipliers);
+    }
+
+    /// @notice Returns task number from when `operator` has been registered.
+    function getFromTaskNumberForOperator(address operator) external view returns (uint32) {
+        return operators[operator].fromTaskNumber;
+    }
+
+    /// @notice Returns the operatorId for the given `operator`
+    function getOperatorId(address operator) external view returns (bytes32) {
+        return operators[operator].operatorId;
+    }
+
+    /// @notice Returns the number of registries
+    function numRegistries() external view returns (uint256) {
+        return registries.length;
+    }
+
+    function registerOperator(address, bytes32, bytes calldata) external override pure {
+        revert("BLSIndexRegistryCoordinator.registerOperator: cannot use overrided StakeRegistry.registerOperator on BLSIndexRegistryCoordinator");
+    }
+
+    /**
+     * @notice Registers the operator with the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registering for
+     * @param registrationData is the data that is decoded to get the operator's registration information
+     */
+    function registerOperator(bytes calldata quorumNumbers, bytes calldata registrationData) external {
+        // get the operator's BLS public key
+        BN254.G1Point memory pubkey = abi.decode(registrationData, (BN254.G1Point));
+        // call internal function to register the operator
+        _registerOperator(msg.sender, quorumNumbers, pubkey);
+    }
+
+    /**
+     * @notice Registers the operator with the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registering for
+     * @param pubkey is the BLS public key of the operator
+     */
+    function registerOperator(bytes calldata quorumNumbers, BN254.G1Point memory pubkey) external {
+        _registerOperator(msg.sender, quorumNumbers, pubkey);
+    }
+
+    function deregisterOperator(address, bytes32, bytes calldata) external override pure {
+        revert("BLSIndexRegistryCoordinator.deregisterOperator: cannot use overrided StakeRegistry.deregisterOperator on BLSIndexRegistryCoordinator");
+    }
```

Comment:
did we decide to leave this function in the interface? or should this function no longer be needed?

Reply:
ah, nevermind, this is because the contract inherits from StakeRegistry. There should be a comment that notes this, at least.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/e9b878193f2dc85ed78710dfcdf79a26bdcfc5d4